File: backend/src/models/zona_critica.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class ZonaCriticaCreate(BaseModel):
    latitude_centro: float
    longitude_centro: float
    raio_metros: int = 500
    nome: Optional[str] = None


class ZonaCriticaResponse(BaseModel):
    zona_id: str
    latitude_centro: float
    longitude_centro: float
    raio_metros: int
    nome: Optional[str] = None
    total_acidentes: int
    acidentes_graves: int
    causa_mais_frequente: Optional[str] = None
    nivel_risco: str
    recomendacao_melhoria: Optional[str] = None
    validado: bool
    created_at: datetime

File: backend/src/models/ZonaCritica.js
const mongoose = require('mongoose');

const zonaCriticaSchema = new mongoose.Schema({
  zona_id: { type: String },
  latitude_centro: { type: Number, required: true },
  longitude_centro: { type: Number, required: true },
  raio_metros: { type: Number, default: 500 },
  nome: { type: String, required: true },
  total_acidentes: { type: Number, default: 0 },
  acidentes_graves: { type: Number, default: 0 },
  causa_mais_frequente: { type: String, default: null },
  nivel_risco: { type: String, default: 'MEDIO' },
  recomendacao_melhoria: { type: String, default: null },
  delimitacoes: [{ lat: Number, lng: Number }],
  tipo_zona: { type: String, enum: ['critica', 'vigilancia', 'segura'], default: 'critica' },
  validado: { type: Boolean, default: false },
  monitores: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }],
  created_at: { type: Date, default: Date.now }
}, { collection: 'zonas_criticas', timestamps: false });

module.exports = mongoose.model('ZonaCritica', zonaCriticaSchema);
